cpp/include/cudf/reduction/detail/reduction_functions.hpp

```diff
@@ -51,6 +51,27 @@ std::unique_ptr<scalar> sum(column_view const& col,
                             rmm::cuda_stream_view stream,
                             rmm::device_async_resource_ref mr);
 
+/**
+ * @brief Computes sum with overflow detection of int64_t elements in input column
+ *
+ * Returns a struct scalar with {sum: int64_t, overflow: bool} fields.
+ * Only supports int64_t input columns.
+ *
+ * @throw cudf::logic_error if input column type is not int64_t
+ *
+ * @param col input column to compute sum with overflow detection (must be int64_t)
+ * @param output_dtype data type of return type (must be struct)
+ * @param init initial value of the sum
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ * @param mr Device memory resource used to allocate the returned scalar's device memory
+ * @return Struct scalar with sum and overflow flag
+ */
+std::unique_ptr<scalar> sum_with_overflow(column_view const& col,
+                                          data_type const output_dtype,
```

Comment:
I recommend to change all `output_dtype` to `output_type` to be consistent with the changes in https://github.com/rapidsai/cudf/pull/19267/.